remote-api/src/main/java/org/alfresco/rest/api/impl/rules/RuleSetsImpl.java

```diff
@@ -69,6 +76,34 @@ public RuleSet getRuleSetById(String folderNodeId, String ruleSetId, List<String
         return ruleSetLoader.loadRuleSet(ruleSetNode, folderNode, includes);
     }
 
+    @Override
+    public RuleSetLink linkToRuleSet(String folderNodeId, String linkToNodeId)
+    {
+
+        final NodeRef folderNodeRef = validator.validateFolderNode(folderNodeId,false);
+        final NodeRef linkToNodeRef = validator.validateFolderNode(linkToNodeId, false);
+
+        //The target node should have pre-existing rules to link to
+        if (!ruleService.hasRules(linkToNodeRef)) {
+            throw new AlfrescoRuntimeException("The target node has no rules to link.");
+        }
+
+        //The folder shouldn't have any pre-existing rules
+        if (ruleService.hasRules(folderNodeRef)) {
+            throw new AlfrescoRuntimeException("Unable to link to a ruleset because the folder has pre-existing rules.");

```

Comment:
Do you know what status code this returns?  I would assume 500, but we want 40x (maybe 400?)

Same comment for the other AlfrescoRuntimeException above.